pagopa/io-app
# feat: [IOCOM-1335,IOCOM-1421] FIMS flow loading screens

## Short description
addition of loading screens to FIMS' flow, also refactored its reducer's data structure

## List of changes proposed in this pull request
- Added required I18n entries
- refactored FIMS state's data structure to better discern between the flow's steps
- small saga update to take ActionProps instead of using a selector
- updated `FimsFlowHandlerScreen` to correctly display the various loading/error screens
- updated selectors to better suite the new data structure

## How to test
check out [this branch of io-dev-server](https://github.com/pagopa/io-dev-api-server/tree/feature/fims), and make sure that when running a full FIMS flow (messages_home -> 1 CTA FIMS SSO -> run flow):
- nothing breaks, or if it does the error is gracefully handled with its own screen
- the first and second loading screens are different, and follow the specified design.
